fitbenchmarking/fitbenchmarking
# 702 Extend linting tests

#### Description of Work

Fixes [#702](https://github.com/fitbenchmarking/fitbenchmarking/issues/702) 

Modify linting_tests.sh to loop over directories to test all code for linting errors

#### Testing Instructions

1. Does linting test run as expected?

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
